### PR TITLE
add infrastructure for handling phone-related messages from gregor

### PR DESCRIPTION
**Pull request by xgess**
see https://github.com/keybase/keybase/pull/3042 for where these messages are coming from

* receive and dismiss the three phone messages from gregor
* fire notifications for each

i made them three separate ones because i figured the UI would want to handle them in potentially very different ways. mainly superseded. if that one comes in, you probably really want to know about it.  but i'm happy to collapse this down into a single notification with the type of change that happened.

### CODE REVIEW

File: go/service/phone_numbers.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+	"github.com/keybase/client/go/gregor"
+	"github.com/keybase/client/go/libkb"
+	gregor1 "github.com/keybase/client/go/protocol/gregor1"
+	keybase1 "github.com/keybase/client/go/protocol/keybase1"
+	"github.com/keybase/go-framed-msgpack-rpc/rpc"
+
+	"golang.org/x/net/context"
+)
+
+type PhoneNumbersHandler struct {
```
Comment (zapu) [inline go/service/phone_numbers.go]: if we have a new file and new type for handling phone numbers, can we also move 
```
func (h *UserHandler) AddPhoneNumber(ctx context.Context, arg keybase1.AddPhoneNumberArg)
func (h *UserHandler) VerifyPhoneNumber(ctx context.Context, arg keybase1.VerifyPhoneNumberArg)
func (h *UserHandler) GetPhoneNumbers(ctx context.Context, sessionID int)
```
here, from `service/user.go`

File: protocol/avdl/keybase1/phone_numbers.avdl [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+    record PhoneNumberVerifiedMsg {
+        @jsonkey("phone")
+        string phoneNumber;
+    }
+
+    record PhoneNumberSupersededMsg {
+        @jsonkey("phone")
+        string phoneNumber;
+    }
```
Comment (zapu) [inline protocol/avdl/keybase1/phone_numbers.avdl]: BTW you can also just do one msg for all 3, like `record PhoneNumberMsg` or something like that. Your call.

Comment (xgess) [reply protocol/avdl/keybase1/phone_numbers.avdl]: yeah i mentioned that in the PR description. i initially did that, and thought the front end might want to treat them very differently (e.g. add phone number, don't care. superseded phone number: notify right away). i'm extremely indifferent.

File: go/service/phone_numbers.go (hunk 1 above)
Comment (xgess) [reply go/service/phone_numbers.go]: excellent idea.

File: go/client/rpc.go [hunk 3]
```diff
@@ -145,6 +145,14 @@ func GetUserClient(g *libkb.GlobalContext) (cli keybase1.UserClient, err error)
 	return
 }
 
+func GetPhoneNumbersClient(g *libkb.GlobalContext) (cli keybase1.PhoneNumbersClient, err error) {
+	var rcli *rpc.Client
+	if rcli, _, err = GetRPCClientWithContext(g); err == nil {
+		cli = keybase1.PhoneNumbersClient{Cli: rcli}
+	}
+	return
+}
```
Comment (zapu) [inline go/client/rpc.go]: man, can you imagine how neat this file could be if we had generics or any sort of templating in go? one can dream

File: go/systests/phone_number_test.go [hunk 4]
```diff
@@ -110,3 +110,97 @@ func TestInvalidPhoneNumberResolve(t *testing.T) {
...
+	require.NoError(t, err)
+	cli2 := &client.CmdVerifyPhoneNumber{
+		Contextified: libkb.NewContextified(ann.tc.G),
+		PhoneNumber:  phone,
+		Code:         code,
+	}
+	err = cli2.Run()
+	require.NoError(t, err)
+	ann.drainGregor()
+	require.Equal(t, annListener.verifiedPhones, expectedNotification)
```
Comment (zapu) [inline go/systests/phone_number_test.go]: maybe one thing if you still have this branch checked out somewhere (if not just hit merge):
I'd add something like 
```
require.Equal(t, annListener.supersededPhones, []keybase1.PhoneNumber{})
```
to ensure other notifications did not sneak in unexpected.

Comment (xgess) [reply go/systests/phone_number_test.go]: 👍

File: go/client/rpc.go (hunk 3 above)
Comment (xgess) [reply go/client/rpc.go]: lol. it would not need to be a whole file.

**Pull request merged**